synapseclient/core/exceptions.py

```diff
@@ -81,6 +81,10 @@ class SynapseUploadFailedException(SynapseError):
     """Raised when an upload failed. Should be chained to a cause Exception"""
 
 
+class SynapseQueryError(SynapseError):
```

Comment:
Discussion: Is this just a "ValueError"?

Reply:
Tbh our usage of exceptions throughout the client is all over the place. I'm not opposed to this new exception, however, I think your right @thomasyu888 that the way it's being used is as a `ValueError`